### PR TITLE
service: ptm: Add TS, nifm: Stub GetInternetConnectionStatus

### PR DESCRIPTION
**Pull request by german77**
The Homebrew menu still fails to boot. But at least those 2 services are implemented. This PR also changes PSM to a subservice of PTM.

### CODE REVIEW

File: src/core/hle/service/nifm/nifm.cpp [hunk 1]
```diff
@@ -30,6 +30,19 @@ enum class RequestState : u32 {
...
+    WiFi = 1,
+    Ethernet = 2,
+};
+
+enum class InternetConnectionStatus : u8 {
+    ConnectingUnknown1,
+    ConnectingUnknown2,
+    ConnectingUnknown3,
+    ConnectingUnknown4,
+    Connected
```
Comment (Morph1984) [inline src/core/hle/service/nifm/nifm.cpp:43]:
```suggestion
    Connected,
```

File: src/core/hle/service/nifm/nifm.cpp [hunk 2]
```diff
@@ -405,13 +425,32 @@ class IGeneralService final : public ServiceFramework<IGeneralService> {
...
+        LOG_WARNING(Service_NIFM, "(STUBBED) called");
+
+        struct Output {
+            InternetConnectionType type{InternetConnectionType::WiFi};
+            u8 wifi_strength{3};
+            InternetConnectionStatus state{InternetConnectionStatus::Connected};
+        };
+        static_assert(sizeof(Output) == 0x3, "Output has incorrect size.");
+
+        const Output out{};
```
Comment (Morph1984) [inline src/core/hle/service/nifm/nifm.cpp:447]:
```suggestion
        constexpr Output out{};
```

File: src/core/hle/service/ptm/psm.h [hunk 3]
```diff
@@ -3,16 +3,29 @@
...
+        LowPowerCharger = 2,
+        Unknown = 3,
+    };
 
-} // namespace Service::PSM
+    void GetBatteryChargePercentage(Kernel::HLERequestContext& ctx);
+    void GetChargerType(Kernel::HLERequestContext& ctx);
+    void OpenSession(Kernel::HLERequestContext& ctx);
+
+    u32 battery_charge_percentage{100}; // 100%
```
Comment (Morph1984) [inline src/core/hle/service/ptm/psm.h:27]: This is self-explanatory

```suggestion
    u32 battery_charge_percentage{100};
```

File: src/core/hle/service/ptm/ts.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+TS::~TS() = default;
+
+void TS::GetTemperature(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp{ctx};
+    const auto location{rp.PopEnum<Location>()};
+
+    LOG_WARNING(Service_HID, "(STUBBED) called. location={}", location);
+
+    const s32 temperatureC = location == Location::Internal ? 35 : 20;
```
Comment (Morph1984) [inline src/core/hle/service/ptm/ts.cpp:34]:
```suggestion
    constexpr s32 temperature = location == Location::Internal ? 35 : 20;
```

Comment (german77) [reply src/core/hle/service/ptm/ts.cpp:34]: How do you intend to make this constexpr if `location` is a runtime variable

Comment (Morph1984) [reply src/core/hle/service/ptm/ts.cpp]: Right, disregard this one

**Review by liamwhite — APPROVED** [review]

**Review by bunnei — APPROVED** [review]

**Pull request merged**